util-tracking/src/main/scala/sbt/util/Tracked.scala

```diff
@@ -293,21 +295,36 @@ class Difference(
   private def cachedFilesInfo = store.read(default = FilesInfo.empty[style.F])(style.formats).files
   private def raw(fs: Set[style.F]): Set[File] = fs.map(_.file)
 
-  def apply[T](files: Set[File])(f: ChangeReport[File] => T): T = {
+  def apply[T](files: Set[File])(
+      f: ChangeReport[File] => T
+  ): T = {
     val lastFilesInfo = cachedFilesInfo
-    apply(files, lastFilesInfo)(f)(_ => files)
+    apply(files, lastFilesInfo)(f)({ case _ => files })
   }
 
-  def apply[T](f: ChangeReport[File] => T)(implicit toFiles: T => Set[File]): T = {
+  def apply[T](files: Set[File], toCache: PartialFunction[T, Set[File]])(
```

Comment:
Would it make sense to give this a different name than `apply`?
The overuse of `apply` around tracked is always confusing to me.

Reply:
Agreed, I just tried to keep it consistent with the existing ones. Would deprecate the existing `apply` and expose all of them under a new explicit pattern be an option?

Reply:
Agreed, I just tried to keep it consistent with the existing ones. Would deprecate the existing apply and expose all of them under a new explicit pattern be an option?

Reply:
I think the default one is ok, but the new one probably should have a name instead of distinguishing them by the parameters.